diff --git a/csrc/xpu/attn/xe_2/fmha_utils.hpp b/csrc/xpu/attn/xe_2/fmha_utils.hpp
index 4b5b47e..4ebefd7 100644
--- a/csrc/xpu/attn/xe_2/fmha_utils.hpp
+++ b/csrc/xpu/attn/xe_2/fmha_utils.hpp
@@ -80,10 +80,10 @@ struct chunk_policy_head192 {
 };

 struct chunk_policy_head256 {
-  using ShapeQK = Shape<_256, _32, _32>;
-  using ShapePV = Shape<_256, _32, _32>;
-  using ShapeOut = Shape<_256, _256>;
-  using SubgroupLayoutQK = Layout<Shape<_32, _1, _1>>;
+  using ShapeQK = Shape<_128, _32, _32>;
+  using ShapePV = Shape<_128, _32, _32>;
+  using ShapeOut = Shape<_128, _256>;
+  using SubgroupLayoutQK = Layout<Shape<_16, _1, _1>>;
 };

 struct chunk_policy_head512 {
